lib/linex/message.ex

```diff
@@ -1,31 +1,64 @@
 defmodule Linex.Message do
+  require Logger
+
+  @reply_url "https://api.line.me/v2/bot/message/reply"
+
+  @callback reply(map(), String.t()) :: :ok | :error
+
   @doc """
   Reply the message users send to the channel.
-  """
-  @spec reply(String.t(), String.t()) :: :ok | {:error, HTTPoison.Error.t()}
-  def reply(message, reply_token) do
-    url = "https://api.line.me/v2/bot/message/reply"
 
-    headers = [
-      {"Content-Type", "application/json"},
-      {"Authorization", "Bearer #{Linex.get_conf(:channel_access_token)}"}
-    ]
+  ## Message examples:
 
-    body = %{
-      "replyToken" => reply_token,
-      "messages" => [
-        %{
-          "type" => "text",
-          "text" => message
+    text_message = %{
+      type: "text",
+      text: "Hello, world!"
+    }
+
+    flex_message = %{
+      type: "flex",
+      altText: "this is a flex message",
+      contents: %{
+        type: "bubble",
+        body: %{
+          type: "box",
+          layout: "vertical",
+          contents: [
+            %{type: "text", text: "hello"},
+            %{type: "text", text: "world"}
+          ]
         }
-      ]
+      }
     }
 
+  ref: https://developers.line.biz/en/reference/messaging-api/#message-objects
+  """
+  @spec reply(map(), String.t(), String.t()) :: :ok | :error
+  def reply(message, reply_token, url \\ @reply_url) when is_map(message) do
+    body = build_request_body(message, reply_token)
+    headers = build_headers()
+
     with {:ok, %HTTPoison.Response{status_code: 200}} <-
            HTTPoison.post(url, Jason.encode!(body), headers) do
       :ok
     else
-      {:error, err} -> {:error, err}
+      error ->
+        Logger.error(fn -> inspect(error) end)
+        :error
```

Comment:
Oh, thank you for logging! 

But why we only `:error` instead of attaching the error detail too?
Maybe the lib consumer might want to handle themselves with detail too?

Reply:
Sounds good krub 👍